co-simulation/bridge/carla_integration/sumo_simulation.py

```diff
@@ -226,6 +228,26 @@ def get_all_associated_signals(self, landmark_id):
             signals.update(self._tls[tlid][program_id].get_associated_signals(landmark_id))
         return signals
 
+    def get_live_state(self, landmark_id):
+        link_index_list = []
+        _tlid = -1
+        for tlid, link_index in self.get_all_associated_signals(landmark_id):
+            _tlid = tlid
+            link_index_list.append(link_index)
+
+        state_string = traci.trafficlight.getRedYellowGreenState(_tlid)
+        state_list = []
+        for link_index in link_index_list:
+            state_list.append(state_string[link_index])
```

Comment:
I think I've seen this before, but is this standard behavior for SUMO? I believe previously I've seen SUMO output traffic signal states as "gggGGGGyyyyR" or something like that, which I understand to be the state of each movement or link (whatever they're called) controlled by that signal is that correct?

I'm not a huge fan of using strings in place of more formal data types, but if that's what SUMO does, then I can live with it. I think this logic does effectively only check the first element in that list though right? This for loop will pull off the first element in the list, and it's going to fall into one of these branches with a return statement?

If it's intended to just look at the first element, you can just replace the for block with state_list[0], if you're actually intending to iterate through the whole list then we might need to rework this logic a bit.

Reply:
That is correct. I've added comments in this function and modified the logic in the recent commit. Thanks!